contracts/Forwarder.sol

```diff
@@ -39,14 +39,14 @@ contract Forwarder is IERC721Receiver, ERC1155Receiver, IForwarder {
       return;
     }
 
-    (bool success, ) = parentAddress.call{ value: value }('');
-    require(success, 'Flush failed');
-
     // NOTE: since we are forwarding on initialization,
     // we don't have the context of the original sender.
     // We still emit an event about the forwarding but set
     // the sender to the forwarder itself
     emit ForwarderDeposited(address(this), value, msg.data);
+
+    (bool success, ) = parentAddress.call{ value: value }('');
+    require(success, 'Flush failed');
   }
```

Comment:
This was to address the point related to CHECK EFFECT INTERACTION PATTERN VIOLATED. The point was that a reentrancy attack can occur when the function in the contract makes an external call to another untrusted contract
before resolving any effects. To minimize the risk of event emission order manipulation, it's advisable to implement the check-effect-interaction pattern as extensively as possible. i.e. emitting events before making external calls. Thinking about this, this is probably needed wherever we are interacting with an unknown address but here as we are interacting with the parentAddress, so this could have been skipped probably